public/js/utils/source.js

```diff
@@ -32,7 +36,29 @@ function isPretty(source: {url: string}): boolean {
   return source.url ? /formatted$/.test(source.url) : false;
 }
 
+/**
+ * Show a source url's filename.
+ * If the source does not have a url, use the source id.
+ */
+function getFilename(source: Source) {
+  const url = source.get("url");
+  if (!url) {
+    const id = source.get("id");
+    const sourceId = id.split("/")[1];
+    return `SOURCE${sourceId}`;
+  }
+
+  let name = url;
+  const m = url.toString().match(/.*\/(.+?\..*$)/);
+  if (m && m.length > 1) {
+    name = m[1];
+  }
+
+  return endTruncateStr(name, 50);
+}
```

Comment:
We need to use this in other places as well, like the call stack, right? Can you add that to this PR, so we can be consistent? The call stack includes the location of the file for each frame.

You can also use `basename` from public/js/utils/path.js instead of the regex matching here.